### ISSUE TITLE
NOT NULL constraint failed: memgpt_recall_memory_agent

### ISSUE THREAD

**Issue by clarkandrew** [cpacker/MemGPT#834]
**Bug**

```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: memgpt_recall_memory_agent.model
[SQL: INSERT INTO memgpt_recall_memory_agent (id, user_id, agent_id, role, text, model, name, tool_calls, tool_call_id, embedding, created_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
```

It's trying to insert the model "NULL" into a NOT NULL constrained field. 

**Setup**
MemGPT installed locally in dev env.

**MemGPT version**
0.2.12

**Installation**
pip install -e '.[local]'

**Setup**
MacOS
iTerm2

**Additional context**
1. I'm running a local model with llama.cpp
2. This error still occurs after setting model to 'model = gpt-4' in config.

**Storage config**
```
[archival_storage]
type = chroma

[recall_storage]
type = sqlite
```
---

If you're not using OpenAI, please provide additional information on your local LLM setup:

**Local LLM details**
- mistral via llama.cpp (nvidia gpu)

**Comment by jimlloyd** [comment]
This is a duplicate of #809 but even though I wrote #809 I think this bug report is more useful.

@cpacker @sarahwooders can we prioritize this fix? @sarahwooders have you tested with llama.cpp? I suspect there may be other similar database bugs, such as this one:

```
> Enter your message: My name is Jim
💭 The user's name has changed from Chad to Jim. I need to update my core memory.
⚡🧠 [function] updating memory with core_memory_append
An exception ocurred when running agent.step(): 
Traceback (most recent call last):
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/main.py", line 325, in run_agent_loop
    new_messages, user_message, skip_next_user_input = process_agent_step(user_message, no_verify)
                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/main.py", line 301, in process_agent_step
    new_messages, heartbeat_request, function_failed, token_warning = memgpt_agent.step(
                                                                      ^^^^^^^^^^^^^^^^^^
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/agent.py", line 544, in step
    raise e
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/agent.py", line 529, in step
    self._append_to_messages(all_new_messages)
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/agent.py", line 278, in _append_to_messages
    self.persistence_manager.append_to_messages(added_messages)
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/persistence_manager.py", line 138, in append_to_messages
    self.recall_memory.insert_many([self.json_to_message(m) for m in added_messages])
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/memory.py", line 341, in insert_many
    self.storage.insert_many(messages)
  File "/Users/jim.lloyd/MemGPT-Project/MemGPT/memgpt/agent_store/db.py", line 362, in insert_many
    self.session.commit()
  File "/Users/jim.lloyd/Library/Caches/pypoetry/virtualenvs/pymemgpt-IcfVAHgr-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 1969, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "/Users/jim.lloyd/Library/Caches/pypoetry/virtualenvs/pymemgpt-IcfVAHgr-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/state_changes.py", line 103, in _go
    self._raise_for_prerequisite_state(fn.__name__, current_state)
  File "/Users/jim.lloyd/Library/Caches/pypoetry/virtualenvs/pymemgpt-IcfVAHgr-py3.11/lib/python3.11/site-packages/sqlalchemy/orm/session.py", line 929, in _raise_for_prerequisite_state
    raise sa_exc.PendingRollbackError(
sqlalchemy.exc.PendingRollbackError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (sqlite3.IntegrityError) NOT NULL constraint failed: memgpt_recall_memory_agent.model
[SQL: INSERT INTO memgpt_recall_memory_agent (id, user_id, agent_id, role, text, model, name, tool_calls, tool_call_id, embedding, created_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('fbb0c322-9cf9-497a-b0df-7bf4b1cfae6b', '00000000-0000-0000-0000-faf6b8704770', '1dbab1a9-5e2b-42dd-8619-5b19d54a0d91', 'system', 'You are MemGPT, the latest version of Limnal Corporation\'s digital companion, developed in 2023.\nYour task is to converse with a user from the pers ... (5851 characters truncated) ... ersonality.\n\nMy name is MemGPT.\nI am kind, thoughtful, and inquisitive.\n</persona>\n<human characters="20/2000">\nFirst name: Chad\nJim\n</human>', None, None, 'null', None, None, '2024-01-16 18:58:00.000000')]
(Background on this error at: https://sqlalche.me/e/20/gkpj) (Background on this error at: https://sqlalche.me/e/20/7s2a)
? Retry agent.step()? (Y/n)
```

**Comment by sarahwooders** [comment]
I believe this is fixed in the current version on main https://github.com/cpacker/MemGPT/blob/main/memgpt/agent_store/db.py#L202

There are a lot of other changes though, so if you install from source be wary there might be bugs

**Comment by sarahwooders** [comment]
This is indeed a result of us not testing these changes with llama.cpp, which we should fix in the future #850
